### ISSUE TITLE
Disable or change sorting of generated keys

### ISSUE THREAD

**Issue by canbaytok** [jf-tech/omniparser#193]
Hello there,

first of I want to thank you for creating this amazing library and making it open source!!

I am currently trying to transform some XML data and noticed that the keys generated in the output file are always sorted alphabetically resulting in confusing scenarios, where keys with incrementing numerical suffixes are out of order:
```
KEY1
KEY10
KEY2
KEY3
KEY4
...
```

I wasn't able to find anything in the documentation about this and searching the code didn't help my either (I found some schema validation methods, where sort occurs, but disabling those didn't change the result)

Is it possible to adjust the sorting behavior or disable it altogether and relying on the order inside the schema?

**Comment by jf-tech** [comment]
@canbaytok please include / attach sample input and sample schema that repro the issue so we can take a look. Scrub any PII info if needed.

**Comment by canbaytok** [comment]
Sorry forgot about that. My simplest example would consist of this input file:
```
<?xml version="1.0" encoding="UTF-8"?>
<FOO>BAR</FOO>
```

and this schema:
```
{
    "parser_settings": {
        "file_format_type": "xml",
        "version": "omni.2.1"
    },
    "transform_declarations": {
        "FINAL_OUTPUT": {
            "object": {
                "KEY2": {
                    "const": "2"
                },
                "KEY10": {
                    "const": "10"
                },
                "KEY1": {
                    "const": "1"
                }
            }
        }
    }
}
```

Executing this results in the following output:
```
{
	"KEY1": "1",
	"KEY10": "1",
	"KEY2": "1"
}
```

As you can see the order of the schema is ignored and the output keys are sorted alphabetically.
After opening this issue I took a deep dive into the code and researched the topic a little more. It seems that this behavior is to be expected, given that the [documentation](https://pkg.go.dev/encoding/json#Marshal) (search for "sort") of the `json.Marshal` function states that map keys will be sorted and since omniparser ultimately relies on this function to generate the JSON string, there is no easy way around this.

What I did find was a [stackoverflow thread](https://stackoverflow.com/questions/48293036/prevent-alphabetically-ordering-json-at-marshal) about the very same issue (`json.Marshal` sorts map keys) and people suggesting to use a user defined struct and corresponding `MarshalJSON` implementation, which takes the map key order into account.

But I've since realized that implementing this into omniparser would probably cost too much time for no real benefit, as the JSON specification does not place any emphasis on the map key order and a JSON with ordered or unordered map keys is the same for a computer reading and interpreting it. It just feels wrong for me (a human) to read the tenth item of a group before the second

I would still love to hear about your thoughts on this, before closing the issue :)

**Comment by jf-tech** [comment]
You are correct there is practically no way for omniparser to achieve this without rolling out our own custom unmarshaler (for keep the order in schema json fields and for injesting json input - omniparser does support json) and custom marshaler for output.

That said, json does support array whose order is preserved. Any chance you can utilize that?

**Comment by canbaytok** [comment]
Well using an array would be an option, but I wouldn't be able to declare different keys for each array element so the whole structure of the output changes and is unfortunately not right for my use case :/

Thank you for taking the time and getting back to me!

**Comment by jf-tech** [comment]
@canbaytok  sorry for re-opening the issue. It's an interesting discussion.

What if you can do something like this:

```
{
    "parser_settings": {
        "file_format_type": "xml",
        "version": "omni.2.1"
    },
    "transform_declarations": {
        "FINAL_OUTPUT": {
            "array": [
                { "object": { "KEY2": { "const": "2" } } },
                { "object": { "KEY10": { "const": "10" } } },
                { "object": { "KEY1": { "const": "1" } } }
            ]
        }
    }
}
```

And its output would be like this:

```
	[
		{
			"KEY2": "2"
		},
		{
			"KEY10": "10"
		},
		{
			"KEY1": "1"
		}
	]
```

So basically creating an array of objects, and each of which has a single field whose field name is the key and field value is the content. Granted, the output structure may not be the ideal you want, but wondering if this would be a workable solution?

**Comment by canbaytok** [comment]
I tried just that to test your previous comment, but realized, that it's not suitable for my use case for two reasons:

1. Reading the value of one of these keys/objects requires looping over the whole array to find the corresponding key and then reading the value.
So with an object like this
```json
{
	"KEY1": "1",
	"KEY10": "1",
	"KEY2": "1"
}
```
I can just query `obj["KEY1"]` and instantly retreive its value, whereas the array requires the correct index followed by the map key `obj[2]["KEY1"]`

2. My use case demands a JSON -> XML conversion after parsing. I am using `github.com/clbanning/mxj` for that.
This JSON:
```json
{
	"KEY1": "1",
	"KEY10": "1",
	"KEY2": "1"
}
```
is turned into this XML (at least the inner structure):
```xml
<FOO>
	<KEY1>1</KEY1>
	<KEY10>10</KEY10>
	<KEY2>2</KEY2>
</FOO>
```

This JSON however:
```json
{
	"FOO": [
		{
			"KEY2": "2"
		},
		{
			"KEY10": "10"
		},
		{
			"KEY1": "1"
		}
	]
}
```

is turned into this XML:
```xml
<FOO>
	<KEY2>2</KEY2>
</FOO>
<FOO>
	<KEY10>10</KEY10>
</FOO>
<FOO>
	<KEY1>1</KEY1>
</FOO>
```

I also tried the conversion using the schema you provided and the output does look like I would expect, but it requires that the input file for the conversion starts with a JSON array instead of an object and my use case just doesn't do that, since the keys for which the order "matters" are always nested inside different objects.

**Comment by jf-tech** [comment]
@canbaytok Fair enough. Sorry but adding order and going with custom marshaler/unmarshaler isn't on our roadmap. Let me know if there is anything else we can help.
